src/include/rimage/cavs/cavs_ext_manifest.h

```diff
@@ -0,0 +1,266 @@
+		/* FRONT_LEFT | FRONT_RIGHT | FRONT_CENTER | LOW_FREQUENCY | BACK_LEFT | BACK_RIGHT */
+		uint32_t channel_5_1 : 1;
+		/* FRONT_LEFT | FRONT_RIGHT | FRONT_CENTER | LOW_FREQUENCY | SIDE_LEFT | SIDE_RIGHT */
+		uint32_t channel_5_1_surround : 1;
+		/* FRONT_LEFT | FRONT_RIGHT | FRONT_CENTER | BACK_LEFT | BACK_RIGHT | FRONT_LEFT_OF_CENTER | FRONT_RIGHT_OF_CENTER */
+		uint32_t channel_7_0 : 1;
+		/* FRONT_LEFT | FRONT_RIGHT | FRONT_CENTER | BACK_LEFT | BACK_RIGHT | SIDE_LEFT | SIDE_RIGHT */
+		uint32_t channel_7_0_surround : 1;
+		/* FRONT_LEFT | FRONT_RIGHT | FRONT_CENTER | LOW_FREQUENCY | BACK_LEFT | BACK_RIGHT | FRONT_LEFT_OF_CENTER | FRONT_RIGHT_OF_CENTER */
+		uint32_t channel_7_1 : 1;
+		/* FRONT_LEFT | FRONT_RIGHT | FRONT_CENTER | LOW_FREQUENCY | BACK_LEFT | BACK_RIGHT | SIDE_LEFT | SIDE_RIGHT */
+		uint32_t channel_7_1_surround : 1;
+		uint32_t reserved : 16;
+	} r;
+};
+
+enum mod_stream_type
+{
+	ePcm = 0,	/* PCM stream */
+	eMp3,	/* MP3 encoded stream */
+	eAac,	/* AAC encoded stream */
+	eMaxStreamType,
+	eStreamTypeInvalid = 0xFF
+};
+
+enum mod_type
+{
+	eBaseFwModule = 0,
+	eMixInModule,
+	eMixOutModule,
+	eCopierModule,
+	ePeakVolModule,
+	eUpDwMixModule,
+	eMuxModule,
+	eSrcModule,
+	eWovModule,
+	eFxModule,
+	eAecModule,
+	eKpbModule,
+	eMicSelectModule,
+	eFxFModule,	/*i.e.SmartAmp */
+	eAudClassModule,
+	eFakeCopierModule,
+	eIoDriver,
+	eWhmModule,
+	eGdbStubModule,
+	eSensingModule,
+	eMaxModule,
+	eModuleInvalid = eMaxModule
+} ;
+
+struct fw_pin_description
+{
+	union mod_pin_caps		caps;
+	enum mod_stream_type		format_type;
+	union mod_sample_rates		sample_rate;
+	union mod_sample_sizes		sample_size;
+	union mod_sample_containers	sample_container;
+	union mod_channel_config	ch_cfg;
+};
+
+struct fw_ext_man_header
+{
+	uint32_t id;
+	uint32_t len;	/* sizeof(Extend Manifest) in bytes */
+	uint16_t version_major;	/* Version of Extended Manifest structure */
+	uint16_t version_minor;	/* Version of Extended Manifest structure */
+	uint32_t num_module_entries;
+};
+
+struct fw_ext_mod_config_header
+{
+	uint32_t ext_module_config_length;	/* sizeof(fw_ext_mod_config_header) in bytes */
+	uint32_t guid[4];	/* Module GUID */
+	uint16_t version_major;	/* Module version */
+	uint16_t version_minor;	/* Module version */
+	uint16_t version_hotfix;	/* Module version */
+	uint16_t version_build;	/* Module version */
+	enum mod_type module_type;
+	uint32_t init_settings_min_size;	/* Minimum size of initialization settings (in bytes) */
+	uint16_t num_scheduling_capabilities;	/* number scheduling capabilities supported by the module */
+	uint16_t num_pin_entries;	/* Number of Pin (inputs + ouptuts) */
+};
+
+struct fw_ext_mod_config
+{
+	struct fw_ext_mod_config_header header;
+	struct mod_scheduling_caps sched_caps;
+	struct fw_pin_description *pin_desc;
+};
+
+struct fw_ext_manifest
+{
+	struct fw_ext_man_header header;
+	int mod_conf_count;
+	struct fw_ext_mod_config ext_mod_config_array[FW_MAX_EXT_MODULE_NUM];
+};
```

Comment:
These will all need packed.

Reply:
@lgirdwood  this structure is not defined by fw spec, just by me to save module information to build manifest later, so we don't need to pack it.

Reply:
For above structures, it can't be packed since adsp_fw_parser can't decode it correctly. And other drivers follow adsp_fw_parser, so we don't packed attribute to be compatible with them

Reply:
manifest.h is only for structure that are in the manifest. Internal only structures should go in internal headers.

This needs to be packed otherwise it will build differently with 64bit and 32bit compilers. So instead of using an ```int``` use a int32_t or int64_t to align with the parser..

Reply:
thanks!